File: plugins/meson-templates/meson_templates/resources/src/hello_window.py
{{include "license.py"}}

from gi.repository import Gtk
from .gi_composites import GtkTemplate

@GtkTemplate(ui='{{appid_path}}/{{ui_file}}')
class {{Prefix}}Window(Gtk.ApplicationWindow):
    __gtype_name__ = '{{Prefix}}Window'

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.init_template()

File: plugins/meson-templates/meson_templates/resources/src/helloWindow.js
const Gio = imports.gi.Gio;
const GLib = imports.gi.GLib;
const Gtk = imports.gi.Gtk;
const Lang = imports.lang;

var {{Prefix}}Window = new Lang.Class({
    Name: '{{Prefix}}Window',
    GTypeName: '{{Prefix}}Window',
    Extends: Gtk.ApplicationWindow,
    Template: 'resource://{{appid_path}}/{{prefix}}Window.ui',

    _init(application) {
        this.parent({
            application,
            default_width: 600,
            default_height: 300,
        });
    },
});

